### PR TITLE
Cluster environment group, increase timeout, error reporting

### PR DESCRIPTION
**Pull request by markgoddard**

### CODE REVIEW

File: library/os_container_infra.py [hunk 1]
```diff
@@ -105,8 +105,8 @@ def connect(self, cloud):
         self.client = Client('1', endpoint_override=False, session=self.cloud.session)
 
     def apply(self):
-        tries = 0
-        fail_after = 100
+        timeout = 3600
```
Comment (brtkwr) [inline library/os_container_infra.py:108]: why are we using absolute time rather than retries?

File: library/os_server_interface.py [hunk 2]
```diff
@@ -72,7 +72,12 @@ def __init__(self, **kwargs):
         self.state = kwargs['state']
     
         self.connect(kwargs['cloud'])
-        self.interfaces = [self.cloud.network.find_network(interface) for interface in kwargs['interfaces']]
+        self.interfaces = []
+        for interface in kwargs['interfaces']:
+            network = self.cloud.network.find_network(interface)
+            if not network:
+                raise Exception("Unable to find network '%s'" % interface)
```
Comment (brtkwr) [inline library/os_server_interface.py:79]: Nice. This was on my to-do list :)

File: templates/inventory-container-infra.j2 [hunk 3]
```diff
@@ -1,6 +1,12 @@
 [openstack]
 localhost   ansible_connection='local'    ansible_python_interpreter='/usr/bin/env python'
 
+{% if os_container_infra_environment_group is not none %}
```
Comment (brtkwr) [inline templates/inventory-container-infra.j2:4]: You'll have to show me how do use this. The purpose is not entirely clear to me yet.

Comment (markgoddard) [reply templates/inventory-container-infra.j2:4]: If you set it to the name of an ansible group, all hosts will be added to that group. There will be more context in a PR to p3-appliances

File: library/os_container_infra.py (hunk 1 above)
Comment (markgoddard) [reply library/os_container_infra.py:108]: It's more accurate, and easier to reason about. We don't really care how many times you asked about the cluster status, we care how long it's taken. Less important is the variability introduced by the duration of API calls, which is not included in the timing with retries.

**Comment by brtkwr** [comment]
I am satisfied with your answers.

**Pull request merged**